core/storage/user/gae_models.py

```diff
@@ -838,6 +914,23 @@ def get_or_create(cls, user_id, collection_id):
         else:
             return cls.create(user_id, collection_id)
 
+    @classmethod
+    def export_data(cls, user_id):
+        """Takeout: Export CollectionProgressModel user-relevant properties.
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+        Returns:
+            dict. The user-relevant properties of CollectionProgressModel
+            in a python dict format.
+        """
+        found_models = cls.get_all().filter(cls.user_id == user_id)
+        user_data = {}
+        for user_model in found_models:
+            user_data[user_model.collection_id] = (
+                user_model.completed_explorations)
```

Comment:
One note: could you please change line 815 to "The list of IDs of explorations .." if that is correct? I think the field is unfortunately named but we can at least fix the comment.

(I assume it actually does represent IDs of explorations?)

Reply:
Agreed, found a file that uses it for IDs.